src/components/container/IntradayHeartRateChart/IntradayHeartRateChart.tsx

```diff
@@ -0,0 +1,102 @@
+
+export interface IntradayChartThreshold {
+    value: number
+    referenceLineColor?: ColorDefinition
+    overThresholdLineColor?: ColorDefinition
+}
+
+export interface IntradayHeartRateChartProps {
+    previewState?: IntradayHeartRatePreviewState,
+    dataSource: DeviceDataV2Namespace,
+    aggregationOption: AggregationOption,
+    aggregationIntervalAmount: number,
+    aggregationIntervalType: "Hours" | "Minutes",
+    lineColor?: ColorDefinition,
+    thresholds?: IntradayChartThreshold[],
+    innerRef?: React.Ref<HTMLDivElement>
+}
+
+export default function IntradayHeartRateChart(props: IntradayHeartRateChartProps) {
+    const [data, setData] = useState<LineChartDataPoint[]>([]);
+    const [axis, setAxis] = useState<LineChartAxis>({ yRange: [0, 200], yIncrement: 30, xIncrement: props.aggregationIntervalAmount });
+    const dateRangeContext = useContext(DateRangeContext);
+    let intervalStart = startOfDay(dateRangeContext?.intervalStart ?? new Date());
+    let intervalEnd = add(intervalStart, { days: 1 });
+    let layoutContext = useContext(LayoutContext);
+    const defaultLineColor = resolveColor(layoutContext.colorScheme, props.lineColor) || '#bbb';
+
+    function initialize() {
+        if (props.previewState) {
+            var data: DeviceDataV2Aggregate[] = [];
+            switch (props.previewState) {
+                case "Default":
+                case "CompleteDataWithThresholds":
+                    data = FullDayData;
+                    break;
+                case "PartialDataWithThresholds":
+                    data = HalfDayData;
+                    break;
+            }
+
+            extractStateData(data);
+        } else {
+            const params: DeviceDataV2AggregateQuery = {
+                type: props.dataSource === "Fitbit" ? "activities-heart-intraday" : "Heart Rate",
+                namespace: props.dataSource,
+                observedAfter: intervalStart.toISOString(),
+                observedBefore: intervalEnd.toISOString(),
+                intervalAmount: props.aggregationIntervalAmount ?? 1,
+                intervalType: props.aggregationIntervalType,
+                aggregateFunctions: [props.aggregationOption]
+            };
+
+            hrIntradayDataProvider(params).then(function (data: DeviceDataV2Aggregate[]) {
+                extractStateData(data);
+            }).catch(function (error) {
+                console.log("error");
+                console.log(error);
+            });
+        }
+    }
+
+    function extractStateData(data: DeviceDataV2Aggregate[]) {
+        let axis: LineChartAxis = { yRange: [0, 200], yIncrement: 30, xIncrement: props.aggregationIntervalAmount ?? 3 };
+        let dataPoints : LineChartDataPoint[] = data.map(transformToDeviceDataV2AggregateReduced);
+        dataPoints = dataPoints.sort((a, b) => a.date.getTime() - b.date.getTime());
+
+        if (dataPoints.length){
+            let yValues: number[] = dataPoints.map(d => d.value);
```

Comment:
What happens if you have a gap in the middle of the day e.g. where the user wasn't wearing their device?

Reply:
@chrisnowak The line will continue without breaks.

Reply:
Right - my point is that is not the right behavior though, right?  Since you need each heart rate data point to be represented at the time for the data point

Reply:
This is corrected, and a story is in place for showing the behavior (line break).